tests/base_test_case.py

```diff
@@ -115,12 +115,16 @@ def assertRaisesRegexp(self, expected_exception, expected_regexp,
             self.fail("Expected exception " + str(expected_exception) +
                       " not raised" + msg)
 
+    @staticmethod
+    def assertLazyMessage(assert_function, actual, expected, msg_func):
+        try:
+            assert_function(actual, expected)
+        except AssertionError:
+            assert_function(actual, expected, msg=msg_func())
```

Comment:
I'm confused as to why you need to call `assert_function()` here. Can't you just call `msg_func()`?